### ISSUE TITLE
Runtime permission error in API 23+

### ISSUE THREAD

**Issue by pa1pal** [fossasia/phimpme-android#59]
App crashes while opening in Marshmallow above devices. Needs to get Runtime permission for READ_EXTERNAL_STORAGE.
```
java.lang.RuntimeException: Unable to resume activity {vn.mbm.phimp.me/vn.mbm.phimp.me.PhimpMe}: java.lang.SecurityException: Permission Denial: reading com.android.providers.media.MediaProvider uri content://media/external/images/thumbnails from pid=26836, uid=10126 requires android.permission.READ_EXTERNAL_STORAGE, or grantUriPermission()
                                                                     at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3121)
                                                                     at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3152)
                                                                     at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2495)
                                                                     at android.app.ActivityThread.-wrap11(ActivityThread.java)
                                                                     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1354)
                                                                     at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                     at android.os.Looper.loop(Looper.java:148)
                                                                     at android.app.ActivityThread.main(ActivityThread.java:5443)
                                                                     at java.lang.reflect.Method.invoke(Native Method)
                                                                     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:728)
                                                                     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:618)
                                                                  Caused by: java.lang.SecurityException: Permission Denial: reading com.android.providers.media.MediaProvider uri content://media/external/images/thumbnails from pid=26836, uid=10126 requires android.permission.READ_EXTERNAL_STORAGE, or grantUriPermission()
                                                                     at android.os.Parcel.readException(Parcel.java:1620)
                                                                     at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:183)
                                                                     at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:135)
                                                                     at android.content.ContentProviderProxy.query(ContentProviderNative.java:421)
                                                                     at android.content.ContentResolver.query(ContentResolver.java:491)
                                                                     at android.content.ContentResolver.query(ContentResolver.java:434)
                                                                     at vn.mbm.phimp.me.newGallery.resumeLocalPhoto(newGallery.java:3407)
                                                                     at vn.mbm.phimp.me.newGallery.onResume(newGallery.java:3242)
```

**Comment by heysadboy** [comment]
@pa1pal Can I work on this issue?

**Comment by pa1pal** [comment]
@heysadboy  Yes. Go ahead.

**Comment by dynamitechetan** [comment]
Actually I fixed this.  When I was writing code for camera PERMISSION,  I asked storage and location permissions too. 
Please Check the PR.

**Comment by heysadboy** [comment]
@dynamitechetan Thanks. Please do inform in advance from next time if you work on a bug collectively to avoid clashes.

**Comment by pa1pal** [comment]
@dynamitechetan Yes you did permissions part in your PR, good job. 
But app in crashing on newGallery.java for Read external storage permission. It needs to be done there as well. Look at the stack trace for more info.

**Comment by pa1pal** [comment]
You can use this [guide](https://gist.github.com/vxhviet/c02f8dc0239259a65ac9#file-runtimePermission-md) to implement the runtime permissions in android the correct way.

**Comment by dynamitechetan** [comment]
@pa1pal a wrote the permission check on the Phipme activity. I tried a fresh install on my friends phone it's not crashing.

**Comment by pa1pal** [comment]
@dynamitechetan on which android version?

**Comment by dynamitechetan** [comment]
Marshamallow and Nougat.

**Comment by pa1pal** [comment]
It is strange. Wait.

**Comment by heysadboy** [comment]
@pa1pal It is crashing on marshmallow. I checked.

**Comment by pa1pal** [comment]
@heysadboy @dynamitechetan implemented runtimepermissions using [this](https://github.com/00ec454/Ask) library. It ask for permissions later, but first crash with the above log.

**Comment by dynamitechetan** [comment]
Okay. Trying to fix this.

**Comment by pa1pal** [comment]
![issue](https://cloud.githubusercontent.com/assets/6936968/23541201/d8f2e334-000b-11e7-8c83-c43496dd0659.gif)

**Comment by optnio00** [comment]
i think you should use 25.0.1+ build toolds.
edit this in gradle files

**Comment by Abhi2424shek** [comment]
The build tools are 25.0.2 @optnio00

**Comment by optnio00** [comment]
ok
Then open Android terminal in android studio and see what is the error and then ask again or google that error

**Comment by heysadboy** [comment]
@opticod The error arises due to the updated google guidelines to explicitly accept the permissions from user. It would be great if you could attach screenshots so we know more. The build tools are 25.0.2, please see if you are working on development branch.

**Comment by dynamitechetan** [comment]
No this error was due in onResume method of NewGallery.java
 Actually this was being called before I checked for permissions.
I wrote a check for that now.

**Comment by Abhi2424shek** [comment]
Which PR @dynamitechetan ?

**Comment by Abhi2424shek** [comment]
@pa1pal isn't this solved?

**Comment by pa1pal** [comment]
@Abhi2424shek it wasn't fixed in #52.
#64 did the fix.
